daemon/containerd/image_prune.go

```diff
@@ -60,6 +61,22 @@ func (i *ImageService) ImagesPrune(ctx context.Context, fltrs filters.Args) (*im
 		return nil, err
 	}
 
+	// Prune leases
+	leaseManager := i.client.LeasesService()
+	pullLeases, err := leaseManager.List(ctx, `labels."moby/action.pull"`)
+	if err != nil {
+		return nil, err
+	}
+	for i, lease := range pullLeases {
+		opts := []leases.DeleteOpt{}
+		if i == len(pullLeases)-1 {
+			opts = append(opts, leases.SynchronousDelete)
+		}
+		if err := leaseManager.Delete(ctx, lease, opts...); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
For a follow up - not sure how feasible that is, but it would be nice if we'd be able to determine how much disk space was freed.